lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -2089,53 +2094,95 @@ diagnoseMatch(ModuleDecl *module, NormalProtocolConformance *conformance,
     break;
   }
 
-  case MatchKind::StaticNonStaticConflict:
-    // FIXME: Could emit a Fix-It here.
-    diags.diagnose(match.Witness, diag::protocol_witness_static_conflict,
-                   !req->isInstanceMember());
+  case MatchKind::StaticNonStaticConflict: {
+    auto witness = match.Witness;
+    auto diag = diags.diagnose(witness, diag::protocol_witness_static_conflict,
+                               !req->isInstanceMember());
+    if (req->isInstanceMember()) {
+      assert(isa<FuncDecl>(witness) || isa<VarDecl>(witness));
+      auto loc = isa<FuncDecl>(witness)
+                     ? cast<FuncDecl>(witness)->getStaticLoc()
+                     : cast<VarDecl>(witness)
+                           ->getParentPatternBinding()
+                           ->getStaticLoc();
+      diag.fixItRemove(loc);
+    } else {
+      diag.fixItInsert(witness->getAttributeInsertionLoc(true),
+                       getTokenText(tok::kw_static));
+    }
     break;
-      
-  case MatchKind::SettableConflict:
-    diags.diagnose(match.Witness, diag::protocol_witness_settable_conflict);
+  }
+
+  case MatchKind::SettableConflict: {
+    auto witness = match.Witness;
+    auto diag =
+        diags.diagnose(witness, diag::protocol_witness_settable_conflict);
+    auto PBD = cast<VarDecl>(witness)->getParentPatternBinding();
+    diag.fixItReplace(PBD->getStartLoc(), getTokenText(tok::kw_var));
```

Comment:
This isn't necessarily correct; `isSettable` also checks read-only computed properties.